seahorn/aws-c-common-stubs/s_emplace_item_override.c

```diff
@@ -0,0 +1,83 @@
+#include <aws/common/hash_table.h>
+#include <aws/common/private/hash_table_impl.h>
+
+#include <seahorn/seahorn.h>
+
+#include <utils.h>
+
+/**
+ * Function: s_emplace_item
+ * replace iterating for empty slot with non-deterministically
+ * finding empty slot
+ */
+
+struct hash_table_entry *s_emplace_item(struct hash_table_state *state,
+                                        struct hash_table_entry entry,
+                                        size_t probe_idx) {
+  AWS_PRECONDITION(hash_table_state_is_valid(state));
+
+  if (entry.hash_code == 0) {
+    AWS_RETURN_WITH_POSTCONDITION(NULL, hash_table_state_is_valid(state));
+  }
+
+#ifdef __SEAHORN__
+  size_t index = nd_size_t();
+  assume(index < state->size);
+  assume(state->slots[index].hash_code == 0);
+  state->slots[index] = entry;
+  size_t empty_slot_idx;
```

Comment:
`empty_slot_idx` is unused or something is missing?

Reply:
just need an address to hold "return value" for `aws_hash_table_has_an_empty_slot`

Reply:
I think this is the case where you removed the call to `empty_slot`, so now it is dead

Reply:
oh you are right, I was looking at the other stub, `empty_slot_idx` here is indeed dead